### PR TITLE
[CI] Initialization of e2e integration tests

### PR DESCRIPTION
**Pull request by leecalcote**
**Description**

This PR is incomplete, but uses a common approach that other adapters are using to perform end-to-end testing.

### CODE REVIEW

**Comment by leecalcote** [comment]
@jbyers19 you might make quick work of the `expected pods` and appropriate test assertions.

**Comment by jbyers19** [comment]
@leecalcote, good stuff. I'll need a bit to grok the tests before making changes. Did you want to merge this PR now with the basic test structure there? Or leave it as WIP until I can update the assertions?

**Comment by leecalcote** [comment]
@jbyers19, oops, I missed your question. Originally, I think that this PR can wait around under you validate the appropriate assertions. @Revolyssup might have carried this work forward in another PR, though (and might have engaged you, I don't remember). @Revolyssup, should we close this PR?

**Comment by leecalcote** [comment]
> @jbyers19, oops, I missed your question. Originally, I think that this PR can wait around under you validate the appropriate assertions. @Revolyssup might have carried this work forward in another PR, though (and might have engaged you, I don't remember). @Revolyssup, should we close this PR?

What do you guys think?

**Comment by Revolyssup** [comment]
@leecalcote  Yes I think we can close

File: .github/workflows/e2etest.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          meshery-component-version: $version
+          meshery-server-version: $(jq '.metadata."meshery-server-version"' data.json)
+          k8s-distro: minikube
+          k8s-version: $(jq '.metadata.k8s_version' data.json)
+          service-mesh: nginx-sm
+          service-mesh-version: $(jq '.metadata.service_mesh_version' data.json)
+          tests:
+            # osm-bootstrap: $(jq '.pods_status."osm-bootstrap"' data.json)
+            # osm-controller: $(jq '.pods_status."osm-controller"' data.json)
+            # osm-injector:  $(jq '.pods_status."osm-injector"' data.json)
```
Comment (jbyers19) [inline .github/workflows/e2etest.yml:95]: Just guessing based off of context, but it looks like this is deploying a mesh and then verifying mesh specific pods are running? If that is the case, then I believe these assertions would be sufficient:
```yaml
nginx-mesh-api: $(jq '.pods_status."nginx-mesh-api"' data.json)
nginx-mesh-metrics: $(jq '.pods_status."nginx-mesh-metrics"' data.json)
spire-server: $(jq '.pods_status."spire-server"' data.json)
spire-agent: $(jq '.pods_status."spire-agent"' data.json)  # these are daemonsets, so there will be 1/node
nats-server: $(jq '.pods_status."nats-server"' data.json)
```

**Pull request closed without merging**